lib/pluginmanager/package.rb

```diff
@@ -0,0 +1,61 @@
+require "logstash/util/compress"
+
+class LogStash::PluginManager::Package < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "compress package as a tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "compress package as a zip file", :default => LogStash::Environment.windows?
+  option "--clean", :flag, "clean up the generated dump of plugins", :default => false
+  option "--override", :flag, "Override a previously generated package file", :default => false
+
+  def execute
+    signal_error("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting") unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+    puts("Packaging plugins for offline usage")
+    if override?
+      File.delete(target_file)
+    else
+      validate_target_file
+    end
```

Comment:
I see this as two different behaviour but if you like it to be together I will just write it there.